File: Desktop/Features/Updater/Services/AppInfoService.cs
using System.Reflection;
using StreamTabula.Features.Updater.Models;

namespace StreamTabula.Features.Updater.Services;

public class AppInfoService
{
    public AppInfoModel AppInfo { get; }

    public AppInfoService()
    {
        var assembly = Assembly.GetExecutingAssembly();

        var appName = assembly.GetName().Name ?? "StreamTabula";

        var infoVersionAttr = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
        var currentVersion = infoVersionAttr?.InformationalVersion ?? "1.0.0";

        AppInfo = new AppInfoModel(currentVersion, appName) {
            AuthorName = "inkliuznyk",
            LinkText = "github.com/yevheniy-hliebov/StreamTabula",
            LinkUrl = "https://github.com/yevheniy-hliebov/StreamTabula",
        };
    }
}

File: Client/lib/features/updater/services/app_info_service.dart
import 'package:package_info_plus/package_info_plus.dart';
import 'package:streamtabula/features/updater/models/app_info.dart';

class AppInfoService {
  late final AppInfo appInfo;

  Future<void> init() async {
    final packageInfo = await PackageInfo.fromPlatform();
    final cleanVersion = packageInfo.version.split('+').first;

    appInfo = AppInfo(appName: "StreamTabula", currentVersion: cleanVersion);
  }
}
